pages/docs/tutorials/native/working-with-klib.md

```diff
@@ -0,0 +1,140 @@
+we create a small library called `utils.kt` which contains the following two functions
+
+```kotlin
+package utils
+
+fun printMessage(message: String) {
+    println("Message: $message")
+}
+
+fun printWarning(warning: String) {
+    println("Warning: $warning")
+}
+```
+
+To create a library from this file, we can use the compiler with the parameter `-produce library` or `-p library` for short:
+
+    konanc utils.kt -p library
+    
+By default, the output of the filename is `library.klib`. We can override it using the `-output` or `-o` parameter:
+
+    konanc utils.kt -p library -o utils
+    
+
+The second way to create a library is using the `cinterop` tool which allows us to create a Kotlin/Native library of an existing
+C library. See the [Interop with C tutorial](interop-with-c.html) on how to accomplish this.
+ 
+## Consuming a Kotlin/Native Library
+
+Now that we have our library, we can use it in our application. In our case this is a simple file named `sample.kt` with the following contents:
+
+```kotlin
+import utils.*
+
+fun main(args: Array<String>) {
+    printWarning("App is about to shut down!")
+}
+```
+
+Notice how we need to import the library on the first line using the `import` statement. 
+
+In order for the compiler to correctly link in the library, we need to pass the library name using the `-library` or `-l` paramter
+
+    konanc sample.kt -l utils
+    
+this would then produce a single executable file with no runtime dependencies.    
+    
+## Using and creating library repositories
+
+Often it would be useful to use the same library across multiple applications. To avoid having various copies of the same library, the Kotlin
+compiler can search for libraries in what's called a library repository. The default repository is usually installed under the folder `~/konanc/klib` and we can 
+add and remove our own libraries with a utility named `klib` that ships as part of the compiler tools.  
+
+### Installing libraries to the default repository
+
+The easiest way to install a library so that it can be later referenced by any application is to use `klib` with the following command:
+
+    klib install utils
+    
+This will copy and extract the necessary files to the proper locations in the default library folder, allowing us to then compile our application and link 
+a specific library without needing to have the `klib` file in the project folder:
+
+    konanc sample.kt -l utils
+
+We can of course also uninstall libraries at any point by issuing the command:
+
+    klib remove utils
+    
+    
+### Installing libraries to custom repositories
+
+In addition to using the default repository, we can also have libraries stored in custom repositories, which can be useful for instance
+if we want to share libraries amongst a group of projects.
+
+In order to do this, we can once again use the command `klib`: 
+
+    klib install utils -repository jetbrains
```

Comment:
I think it would be more clear to show an example with a full path to the repository.

Reply:
I've re-worded this to make it less cryptic and more straightforward